satellite/satellitedb/nodeselection.go

```diff
@@ -236,11 +241,12 @@ func nodeSelectionCondition(ctx context.Context, criteria *overlay.NodeCriteria,
 //      SELECT * FROM ($selection WHERE $condition ORDER BY $orderBy, RANDOM()) filtered ORDER BY RANDOM() LIMIT $limit
 //
 type partialQuery struct {
-	selection string
-	condition condition
-	distinct  bool
-	orderBy   string
-	limit     int
+	selection  string
+	condition  condition
+	distinct   bool
+	orderBy    string
+	limit      int
+	aostClause asOfSystemTimeClause
```

Comment:
I would pass in the string rather than the struct, It'll keep the code slightly simpler.